components/browser/state/src/main/java/mozilla/components/browser/state/reducer/BrowserStateReducer.kt

```diff
@@ -68,13 +68,17 @@ internal object BrowserStateReducer {
 }
 
 /**
- * Finds the corresponding tab in the [BrowserState] and replaces it using [update].
+ * Finds the corresponding tab or custom tab in the [BrowserState] and replaces it using [update].
+ *
+ * Consider using the specialized [updateTabState] or [updateCustomTabState] to limit the tabs to be updated
+ * if the properties you want changed are not common to both [SessionState] implementations.
+ *
  * @param tabId ID of the tab to change.
  * @param update Returns a new version of the tab state. Must be the same class,
  * preferably using [SessionState.createCopy].
  */
 @Suppress("Unchecked_Cast")
-internal fun BrowserState.updateTabState(
+internal fun BrowserState.updateAnySessionState(
```

Comment:
We spoke offline but let's rename to `updateTabOrCustomTabState` in line with our selectors.